src/Markup/Avalonia.Markup/Data/Plugins/MethodAccessorPlugin.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Avalonia.Data;
+using System.Reflection;
+using System.Linq;
+
+namespace Avalonia.Markup.Data.Plugins
+{
+    class MethodAccessorPlugin : IPropertyAccessorPlugin
+    {
+        public bool Match(object obj, string methodName)
+            => obj.GetType().GetRuntimeMethods().Any(x => x.Name == methodName);
+
+        public IPropertyAccessor Start(WeakReference reference, string methodName)
+        {
+            Contract.Requires<ArgumentNullException>(reference != null);
+            Contract.Requires<ArgumentNullException>(methodName != null);
+
+            var instance = reference.Target;
+            var method = instance.GetType().GetRuntimeMethods().FirstOrDefault(x => x.Name == methodName);
+
+            if (method != null)
+            {
+                if (method.GetParameters().Length + (method.ReturnType == typeof(void) ? 0 : 1) > 8)
+                {
+                    var exception = new ArgumentException("Cannot create a binding accessor for a method with more than 8 parameters or more than 7 parameters if it has a non-void return type.", nameof(method));
+                    return new PropertyError(new BindingNotification(exception, BindingErrorType.Error));
```

Comment:
7/8 parameters are as large as the `System.Action` and `System.Func` delegates support and we need a concrete delegate type for actually creating the delegate. We could add support for more in Avalonia if we want, but because of the lack of generic parameters there will always be a limit.

Reply:
:+1: makes sense.